javascript/javascript1/week2/index.js

```diff
@@ -129,4 +130,53 @@ addStudentToClass("student5");
 addStudentToClass("queen");
 getNumberOfStudents()
 
-console.dir(class07Students);
\ No newline at end of file
+
+//===================================================================================================================
+// Candy helper optional
+
+const boughtCandyPrices = [];
+const candyPrice = {
+    Sweet: 0.5,
+    Chocolate: 0.7,
+    Toffee: 1.1,
+    Chewinggum: 0.03,
+}
+
+const amountToSpend = Math.random() * 100;
+
+function addCandy(candyType, weight) {
+    do {
+        boughtCandyPrices.push(candyPrice[candyType] * weight);
+        console.log(boughtCandyPrices);
+    }
+    while (canBuyMoreCandy == true) {}
+}
+
+addCandy("Toffee", 20);
+addCandy("Chocolate", 10);
+addCandy("Sweet", 10);
+
+
+function canBuyMoreCandy() { //returns a boolean if the user can buy more candy using the "boughtCandy" array
+    let currentAmount = 0;
+    for (let i in boughtCandyPrices) {
+        currentAmount += boughtCandyPrices[i];
+    }
+
+    if (currentAmount > amountToSpend) {
+        console.log("Enough candy for you!");
+        return false
+    } else {
+        console.log("You can buy more, so please do!");
+        return true
+    }
+    // do {
+    //     console.log("Let us buy more sweets!");
+    //     return true
+    // }
+    // while (amountToSpend > currentAmount) {
+    //     return false;
+    // }
```

Comment:
I suppose, you can buy more candy only if you have enough money to spend. So first check with the canBuyMoreCandy function and then do the array push :) Your way also works if you think otherwise.